content/zh-cn/docs/tasks/administer-cluster/kubeadm/change-package-repository.md

```diff
@@ -299,11 +299,11 @@ version.
    ```
    [kubernetes]
    name=Kubernetes
-   baseurl=https://pkgs.k8s.io/core:/stable:/v{{< param "version" >}}/rpm/
+   baseurl=https://pkgs.k8s.io/core:/stable:/{{< param "version" >}}/rpm/
    enabled=1
    gpgcheck=1
-   gpgkey=https://pkgs.k8s.io/core:/stable:/v{{< param "version" >}}/rpm/repodata/repomd.xml.key
-   exclude=kubelet kubeadm kubectl cri-tools kubernetes-cni
+   gpgkey=https://pkgs.k8s.io/core:/stable:/{{< param "version" >}}/rpm/repodata/repomd.xml.key
+   #exclude=kubelet kubeadm kubectl cri-tools kubernetes-cni
```

Comment:
Why this change?

Reply:
[root@web ~]# cat <<EOF | sudo tee /etc/yum.repos.d/kubernetes.repo
> [kubernetes]
> name=Kubernetes
> baseurl=https://pkgs.k8s.io/core:/stable:/v1.29/rpm/
> enabled=1
> gpgcheck=1
> gpgkey=https://pkgs.k8s.io/core:/stable:/v1.29/rpm/repodata/repomd.xml.key
> exclude=kubelet kubeadm kubectl cri-tools kubernetes-cni
> EOF
[kubernetes]
name=Kubernetes
baseurl=https://pkgs.k8s.io/core:/stable:/v1.29/rpm/
enabled=1
gpgcheck=1
gpgkey=https://pkgs.k8s.io/core:/stable:/v1.29/rpm/repodata/repomd.xml.key
exclude=kubelet kubeadm kubectl cri-tools kubernetes-cni
[root@web ~]# yum install kubelet kubeadm kubectl
Loaded plugins: fastestmirror
Repository base is listed more than once in the configuration
Repository extras is listed more than once in the configuration
Repository updates is listed more than once in the configuration
Loading mirror speeds from cached hostfile
 * base: mirrors.aliyun.com
 * elrepo: mirrors.tuna.tsinghua.edu.cn
 * extras: mirrors.aliyun.com
 * updates: mirrors.aliyun.com
base                                                                                                                      | 3.6 kB  00:00:00     
elrepo                                                                                                                    | 3.0 kB  00:00:00     
epel                                                                                                                      | 4.7 kB  00:00:00     
extras                                                                                                                    | 2.9 kB  00:00:00     
kubernetes                                                                                                                | 1.7 kB  00:00:00     
updates                                                                                                                   | 2.9 kB  00:00:00     
(1/4): epel/x86_64/updateinfo                                                                                             | 1.0 MB  00:00:00     
(2/4): elrepo/primary_db                                                                                                  | 367 kB  00:00:00     
(3/4): epel/x86_64/primary_db                                                                                             | 7.0 MB  00:00:00     
(4/4): kubernetes/primary                                                                                                 | 3.8 kB  00:00:01     
kubernetes                                                                                                                                 25/25
No package kubelet available.
No package kubeadm available.
No package kubectl available.
Error: Nothing to do

Reply:
[root@web ~]# cat <<EOF | sudo tee /etc/yum.repos.d/kubernetes.repo
[kubernetes]
name=Kubernetes
baseurl=https://pkgs.k8s.io/core:/stable:/v1.29/rpm/
enabled=1
gpgcheck=1
gpgkey=https://pkgs.k8s.io/core:/stable:/v1.29/rpm/repodata/repomd.xml.key
#exclude=kubelet kubeadm kubectl cri-tools kubernetes-cni
EOF
[kubernetes]
name=Kubernetes
baseurl=https://pkgs.k8s.io/core:/stable:/v1.29/rpm/
enabled=1
gpgcheck=1
gpgkey=https://pkgs.k8s.io/core:/stable:/v1.29/rpm/repodata/repomd.xml.key
#exclude=kubelet kubeadm kubectl cri-tools kubernetes-cni
[root@web ~]# yum install kubelet kubeadm kubectl
Loaded plugins: fastestmirror
Repository base is listed more than once in the configuration
Repository extras is listed more than once in the configuration
Repository updates is listed more than once in the configuration
Loading mirror speeds from cached hostfile
 * base: mirrors.aliyun.com
 * elrepo: mirrors.tuna.tsinghua.edu.cn
 * extras: mirrors.aliyun.com
 * updates: mirrors.aliyun.com
kubernetes                                                                                                                | 1.7 kB  00:00:00     
Resolving Dependencies
--> Running transaction check
---> Package kubeadm.x86_64 0:1.29.0-150500.1.1 will be installed
--> Processing Dependency: kubernetes-cni >= 1.2.0 for package: kubeadm-1.29.0-150500.1.1.x86_64
--> Processing Dependency: cri-tools >= 1.28.0 for package: kubeadm-1.29.0-150500.1.1.x86_64
---> Package kubectl.x86_64 0:1.29.0-150500.1.1 will be installed
---> Package kubelet.x86_64 0:1.29.0-150500.1.1 will be installed
--> Processing Dependency: socat for package: kubelet-1.29.0-150500.1.1.x86_64
--> Processing Dependency: conntrack for package: kubelet-1.29.0-150500.1.1.x86_64
--> Running transaction check
---> Package conntrack-tools.x86_64 0:1.4.4-7.el7 will be installed
--> Processing Dependency: libnetfilter_cttimeout.so.1(LIBNETFILTER_CTTIMEOUT_1.1)(64bit) for package: conntrack-tools-1.4.4-7.el7.x86_64
--> Processing Dependency: libnetfilter_cttimeout.so.1(LIBNETFILTER_CTTIMEOUT_1.0)(64bit) for package: conntrack-tools-1.4.4-7.el7.x86_64
--> Processing Dependency: libnetfilter_cthelper.so.0(LIBNETFILTER_CTHELPER_1.0)(64bit) for package: conntrack-tools-1.4.4-7.el7.x86_64
--> Processing Dependency: libnetfilter_queue.so.1()(64bit) for package: conntrack-tools-1.4.4-7.el7.x86_64
--> Processing Dependency: libnetfilter_cttimeout.so.1()(64bit) for package: conntrack-tools-1.4.4-7.el7.x86_64
--> Processing Dependency: libnetfilter_cthelper.so.0()(64bit) for package: conntrack-tools-1.4.4-7.el7.x86_64
---> Package cri-tools.x86_64 0:1.29.0-150500.1.1 will be installed
---> Package kubernetes-cni.x86_64 0:1.3.0-150500.1.1 will be installed
---> Package socat.x86_64 0:1.7.3.2-2.el7 will be installed
--> Running transaction check
---> Package libnetfilter_cthelper.x86_64 0:1.0.0-11.el7 will be installed
---> Package libnetfilter_cttimeout.x86_64 0:1.0.0-7.el7 will be installed
---> Package libnetfilter_queue.x86_64 0:1.0.2-2.el7_2 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

=================================================================================================================================================
 Package                                   Arch                      Version                                 Repository                     Size
=================================================================================================================================================
Installing:
 kubeadm                                   x86_64                    1.29.0-150500.1.1                       kubernetes                    9.7 M
 kubectl                                   x86_64                    1.29.0-150500.1.1                       kubernetes                     10 M
 kubelet                                   x86_64                    1.29.0-150500.1.1                       kubernetes                     19 M
Installing for dependencies:
 conntrack-tools                           x86_64                    1.4.4-7.el7                             base                          187 k
 cri-tools                                 x86_64                    1.29.0-150500.1.1                       kubernetes                    8.2 M
 kubernetes-cni                            x86_64                    1.3.0-150500.1.1                        kubernetes                    6.7 M
 libnetfilter_cthelper                     x86_64                    1.0.0-11.el7                            base                           18 k
 libnetfilter_cttimeout                    x86_64                    1.0.0-7.el7                             base                           18 k
 libnetfilter_queue                        x86_64                    1.0.2-2.el7_2                           base                           23 k
 socat                                     x86_64                    1.7.3.2-2.el7                           base                          290 k

Transaction Summary
=================================================================================================================================================
Install  3 Packages (+7 Dependent packages)

Total download size: 54 M
Installed size: 297 M
Is this ok [y/d/N]: 


`#exclude=kubelet kubeadm kubectl cri-tools kubernetes-cni`

Reply:
#exclude=kubelet kubeadm kubectl cri-tools kubernetes-cni 
 I can install kubeadm kubectl kubelet

Reply:
OK, but you're missing the context.

See https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/#call-kubeadm-upgrade for why these `exclude` entries are OK.